src/write.jl

```diff
@@ -64,11 +64,18 @@ end
                 if col isa DictEncode
                     id = dictid[]
                     dictid[] += 1
-                    values = unique(col)
-                    dictencodings[i] = (id, encodingtype(length(values)), values)
+                    refpool = DataAPI.refpool(col.data)
+                    if refpool !== nothing
+                        values = refpool
+                        IT = eltype(DataAPI.refarray(col.data))
```

Comment:
Note that `?DataAPI.refpool` currently doesn't guaranteed that you can call `eltype` on the result. The only guarantee is that "`refpool(A)[refarray(A)[I...]]` must be equal to `A[I...]`." Cf. https://github.com/JuliaData/DataAPI.jl/pull/1/#issuecomment-626141115. Maybe it's time to make the docstring more specific?

Specifically, for CategoricalArrays, it would be quite easy to return a `CategoricalPool` that supports `getindex`, but nothing else. But that wouldn't be very useful and would break this PR. :-p